### PR TITLE
Correctly handle variant specific operations in DAG.

### LINKED ISSUE #48: [BUG] Validation and Env File Gen not Limited to Variants
When validation is taking place and when the conda env files are being generated, the full tree is being used, which leads to non-variant packages ending up in the conda env file.

### PR DESCRIPTION
**Pull request by bethune-bryant**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [x] Did you write any [tests](https://github.com/open-ce/open-ce/blob/main/tests/) to validate this change?  

## Description

Fixes #48

### CODE REVIEW

File: open_ce/build_env.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def build_env(args):
 
     # If repository_folder doesn't exist, create it
     if args.repository_folder and not os.path.exists(args.repository_folder):
-        os.mkdir(args.repository_folder)
+        os.makedirs(args.repository_folder, exist_ok=True)
```
Comment (bnemanich) [inline open_ce/build_env.py:100]: Can remove the exists check in the if statement now.

Comment (bethune-bryant) [reply open_ce/build_env.py:100]: Ah, yeah. I made this change because it would fail if the parent folder didn't exist.

Comment (bethune-bryant) [reply open_ce/build_env.py:100]: I went ahead and updated the other locations too.

**Review by jayfurmanek — APPROVED** [review]
This looks good.

**Review by bnemanich — APPROVED** [review]

**Pull request merged**